rfcs/2021-02-04-5802-event-metadata.md

```diff
@@ -0,0 +1,206 @@
+# RFC 5802 - 2021-02-04 - Event Metadata
+
+This RFC introduces a plan to associate persistent metadata with every Vector event, both logs and metrics.
+
+## Scope
+
+This RFC will cover the placement of the metadata, its handling, and
+some potential contents. It will not comprehensively cover the specific
+contents beyond what is required for the initial task, as that will
+depend on implement requirements of the relevant features.
+
+## Motivation
+
+There are two upcoming features planned for Vector that require events
+to be associated with additional data that must persist throughout the
+life of an event:
+
+1. End-to-end latency measurement
+2. End-to-end acknowledgements
+
+Both of these features require each event to be tagged with several
+pieces of data. These datum are not part of the data read from the
+source, nor part of what is sent to the sinks, but exist outside of the
+normal stream.
```

Comment:
I am wondering about how we could let people access the metadata when desired. For example, they might like to have the ingestion timestamp added to the event itself. We can leave that as future work though.

Reply:
I don't have any real objection to adding read access to the metadata, though it isn't core to this RFC.

Reply:
Well, things that need to show up in the event will still need to be added to the event data. However, anything else is still up for review.